src/Integration.Vsix/Analysis/DisableRuleCommand.cs

```diff
@@ -156,23 +156,35 @@ private void Execute(object sender, EventArgs e)
             SonarRuleRepoKeys.TypeScript
         };
 
-        private static void CalculateStatuses(SonarCompositeRuleId rule, SonarLintMode mode, out bool isVisible, out bool isEnabled)
+        // For Sonar rules the command is always visible but can be enabled (Standalone mode)/ disabled (Connected mode).
+        // For Non Sonar rules the command is always invisible and disabled.
+        private static void CalculateStatuses(SonarCompositeRuleId rule, ISonarQubeService sonarQubeService, out bool isVisible, out bool isEnabled)
         {
             // Special case: Secrets
-            // We don't support connected mode for Secrets at the moment so we allow disabling
-            // secrets rules in connected mode.
+            // We don't support connected mode for Secrets in SonarQubeVersions < 9.9 so we allow disabling
+            // secrets rules in those cases.
             if (SonarRuleRepoKeys.AreEqual(SonarRuleRepoKeys.Secrets, rule.RepoKey))
             {
                 isVisible = true;
-                isEnabled = true;
+
```

Comment:
I don't like this very much but I tried a few different options and all of them were equally ugly.

Reply:
Can't we use the new `IConnectedModeSecrets` service you created? It should have the correct logic.